online-shop/src/app/components/containers/products-list/products-list.component.ts

```diff
@@ -1,20 +1,38 @@
-import { Component, OnInit } from '@angular/core';
-import { Product } from 'src/app/modules/shared/types/products.types';
+import { Component, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+import { Product } from 'src/app/modules/shared/types/product.types';
+import { User } from 'src/app/modules/shared/types/user.types';
+import { AuthService } from 'src/app/services/auth.service';
 import { ProductService } from 'src/app/services/products.service';
 
 @Component({
   selector: 'app-products-list',
   templateUrl: './products-list.component.html',
   styleUrls: ['./products-list.component.scss']
 })
-export class ProductsListComponent implements OnInit {
+export class ProductsListComponent implements OnInit{
   products: Product[] = [];
-  constructor(private service: ProductService) { }
-
+  user!:User;
+  constructor(private service: ProductService, private authenticationService:AuthService) { }
   ngOnInit(): void {
     this.getProducts();
+    this.authenticationService.getUserProfile().subscribe(u=>
+      {this.user=u;
+      console.log(u);
+      localStorage.setItem('userId',this.user.id!);
+      let isAdmin='no';
+      this.user.roles.forEach(role=>{
+        if(role=='admin'){
```

Comment:
In JavaScript equality check is different from other languages. Mosttly, in every place where you have to check the equality, you have to use ===. Check the difference here: https://www.freecodecamp.org/news/loose-vs-strict-equality-in-javascript/

Check for logs and commented code.